core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorFactory.java

```diff
@@ -14,13 +14,31 @@
 package io.trino.spi.connector;
 
 import com.google.errorprone.annotations.CheckReturnValue;
+import io.trino.spi.Experimental;
 
 import java.util.Map;
+import java.util.Set;
 
 public interface ConnectorFactory
 {
     String getName();
 
     @CheckReturnValue
     Connector create(String catalogName, Map<String, String> config, ConnectorContext context);
+
+    /**
+     * Returns property names that may contain security-sensitive information.
+     * In addition to properties that are explicitly known to the connector as
+     * security-sensitive, it may also return properties that are unknown or unused.
+     * In other words, if the connector cannot classify a property, it treats it as
```

Comment:
```suggestion
     * In other words, if the connector cannot classify a property, it SHOULD/MUST it as
```

IMO "should"?

Reply:
Yes, I think this should be a guideline, not a strict rule. Ultimately, it is up to the connector to decide how to handle this case.